lxd/device/gpu_physical.go

```diff
@@ -98,10 +101,202 @@ func (d *gpuPhysical) Start() (*deviceConfig.RunConfig, error) {
 	return d.startContainer()
 }
 
+// startCDIDevices starts all the devices given in a CDI specification:
+// * `unix-char` (representing the card and non-card devices)
+// * `disk` (representing the mounts)).
+func (d *gpuPhysical) startCDIDevices(cdiID cdi.ID, configDevices cdi.ConfigDevices, runConf *deviceConfig.RunConfig) error {
+	srcFDHandlers := make([]*os.File, 0)
+	defer func() {
+		for _, f := range srcFDHandlers {
+			_ = f.Close()
+		}
+	}()
+
+	for _, conf := range configDevices.UnixCharDevs {
+		if conf["source"] == "" {
+			return fmt.Errorf("The source of the unix-char device %q used for CDI is empty", conf)
+		}
+
+		_, major, minor, err := unixDeviceAttributes(conf["source"])
+		if err != nil {
+			return err
+		}
+
+		// Here putting a 'cdi.unix' prefix with 'd.name' as a device name with create an directory entry like:
+		// <lxd_var_path>/devices/<instance_name>/cdi.unix.<gpu_device_name>.<path_encoded_relative_dest_path>
+		// 'unixDeviceSetupCharNum' is already checking for dupe entries so we have no validation to do here.
+		err = unixDeviceSetupCharNum(d.state, d.inst.DevicesPath(), "cdi.unix", d.name, conf, major, minor, conf["path"], false, runConf)
```

Comment:
Can you make cdi.unix and cdi.disk a constant, like we do for infiniband, and then use them in place of hardocoded strings please?

Reply:
Because this is CDI related, should them be defined in the `cdi` package?

Reply:
i would keep in this device package because its not related to the cdi spec generation, but only to how LXD devices use that info on the LXD side.

Reply:
fixed